src/vs/workbench/contrib/tags/electron-browser/workspaceTagsService.ts

```diff
@@ -468,10 +512,70 @@ export class WorkspaceTagsService implements IWorkspaceTagsService {
 					// Ignore errors when resolving file or parsing file contents
 				}
 			});
-			return Promise.all([...packageJsonPromises, ...requirementsTxtPromises, ...pipfilePromises]).then(() => tags);
+
+			const pomPromises = getFilePromises('pom.xml', this.fileService, this.textFileService, content => {
+				try {
+					let dependenciesContent;
+					while (dependenciesContent = dependenciesRegex.exec(content.value)) {
+						let dependencyContent;
+						while (dependencyContent = dependencyRegex.exec(dependenciesContent[1])) {
+							const groupIdContent = groupIdRegex.exec(dependencyContent[1]);
+							const artifactIdContent = artifactIdRegex.exec(dependencyContent[1]);
+							if (groupIdContent && artifactIdContent) {
+								this.tagJavaDependency(groupIdContent[1], artifactIdContent[1], 'workspace.pom.', tags);
+							}
+						}
+					}
+				}
+				catch (e) {
+					// Ignore errors when resolving maven dependencies
+				}
+			});
+
+			const gradlePromises = getFilePromises('build.gradle', this.fileService, this.textFileService, content => {
+				try {
+					this.processGradleDependencies(content.value, gradleLooseRegex, tags);
+					this.processGradleDependencies(content.value, gradleCompactRegex, tags);
+				}
+				catch (e) {
+					// Ignore errors when resolving gradle dependencies
+				}
+			});
+
+			const androidPromises = (folders as URI[]).map(workspaceUri => {
+				const manifest = workspaceUri.with({ path: `${workspaceUri.path !== '/' ? workspaceUri.path : ''}/app/src/main/AndroidManifest.xml` });
+				return this.fileService.exists(manifest).then(result => {
+					if (result) {
+						tags['workspace.java.android'] = true;
+					}
+				}, err => {
+					// Ignore errors when resolving android
+				});
+			});
+			return Promise.all([...packageJsonPromises, ...requirementsTxtPromises, ...pipfilePromises, ...pomPromises, ...gradlePromises, ...androidPromises]).then(() => tags);
 		});
 	}
 
+	private processGradleDependencies(content: string, regex: RegExp, tags: Tags): void {
+		let dependencyContent;
+		while (dependencyContent = regex.exec(content)) {
+			const groupId = dependencyContent[1];
+			const artifactId = dependencyContent[2];
+			if (groupId && artifactId) {
+				this.tagJavaDependency(groupId, artifactId, 'workspace.gradle.', tags);
+			}
+		}
+	}
+
+	private tagJavaDependency(groupId: string, artifactId: string, prefix: string, tags: Tags): void {
+		for (const javaLibrary of JavaLibrariesToLookFor) {
+			if ((groupId === javaLibrary.groupId || new RegExp(javaLibrary.groupId).test(groupId)) &&
+				(artifactId === javaLibrary.artifactId || new RegExp(javaLibrary.artifactId).test(artifactId))) {
+				tags[prefix + javaLibrary.tag] = true;
```

Comment:
Can mutating tags be avoided by return the result?

Reply:
Fixed.